internal/provision/devices.go

```diff
@@ -130,3 +89,129 @@ func LoadDevices(path string, dic *di.Container) errors.EdgeX {
 
 	return nil
 }
+
+func loadDevicesFromFile(path string, serviceName string, lc logger.LoggingClient) ([]requests.AddDeviceRequest, errors.EdgeX) {
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return []requests.AddDeviceRequest{}, errors.NewCommonEdgeX(errors.KindServerError, "failed to create absolute path for devices", err)
+	}
+
+	files, err := os.ReadDir(absPath)
+	if err != nil {
+		return []requests.AddDeviceRequest{}, errors.NewCommonEdgeX(errors.KindServerError, "failed to read directory for devices", err)
+	}
+
+	if len(files) == 0 {
+		return []requests.AddDeviceRequest{}, nil
+	}
+
+	lc.Infof("Loading pre-defined devices from %s(%d files found)", absPath, len(files))
+	var addDevicesReq, processedDevicesReq []requests.AddDeviceRequest
+	for _, file := range files {
+		fullPath := filepath.Join(absPath, file.Name())
+		processedDevicesReq = processDevices(fullPath, fullPath, serviceName, nil, lc)
+		if len(processedDevicesReq) > 0 {
+			addDevicesReq = append(addDevicesReq, processedDevicesReq...)
+		}
+	}
+	return addDevicesReq, nil
+}
+
+func loadDevicesFromUri(inputUri string, serviceName string, secretProvider interfaces.SecretProvider, lc logger.LoggingClient) ([]requests.AddDeviceRequest, errors.EdgeX) {
+	parsedUrl, err := url.Parse(inputUri)
```

Comment:
Can just pass in displayPath rather than having to to the parse/error handling again.